src/rmind/components/objectives/forward_dynamics.py

```diff
@@ -47,12 +47,12 @@ def __init__(
     ) -> None:
         super().__init__()
 
-        self.encoder = encoder
+        self.encoder = encoder  # ty:ignore[unresolved-attribute]
```

Comment:
it seems annotating `encoder`, etc resolves this

Reply:
do we still need to decorate corresponding classes with `@final`?